### PR TITLE
SSCS-10523 bump and add SOR doc type

### PR DESCRIPTION
**Pull request by nickhill111**
https://tools.hmcts.net/jira/browse/SSCS-10523

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java [hunk 1]
```diff
@@ -128,21 +132,25 @@ private static void handlePostHearing(SscsCaseData sscsCaseData) {
...
+                case STATEMENT_OF_REASONS:
+                    return sscsCaseData.getDocumentGeneration().getStatementOfReasonsBodyContent();
+                case PERMISSION_TO_APPEAL:
+                case LIBERTY_TO_APPLY:
+                default:
+                    break;
+            }
         }
+
+        throw new IllegalArgumentException("handlePostHearing has unexpected postHearingRequestType: " + postHearingRequestType);
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java:153]: I don't think this change is required. The exception should be in the default case, and null type will be covered. Remember the exception needs to be thrown if one of the remaining post hearing types is implemented, but this method is not updated appropriately.

File: src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java [hunk 2]
```diff
@@ -170,21 +178,45 @@ private static DocumentLink getPreviewDocument(
...
+                    return DocumentType.CORRECTION_APPLICATION;
+                case STATEMENT_OF_REASONS:
+                    return DocumentType.STATEMENT_OF_REASONS_APPLICATION;
+                default:
+                    break;
+            }
         }
-        return documentType;
+
+        throw new IllegalArgumentException("Unexpected request type: " + postHearingRequestType);
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java:195]: As above, put the exception back to the default case

**Review by amoslv1 — CHANGES_REQUESTED** [review]
Some suggestions

File: src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java (hunk 2 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java:195]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java (hunk 1 above)
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java:153]: done

**Review by amoslv1 — DISMISSED** [review]

**Review by amoslv1 — APPROVED** [review]

**Review by amoslv1 — APPROVED** [review]

**Review by amoslv1 — DISMISSED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Pull request merged**